### DISCUSSION TITLE
Print jobs get canceled at printer

### DISCUSSION THREAD

**Discussion by mcp292** [OpenPrinting/cups#1090] [Q&A]
# Purpose

This is an attempt to better document and organize what started in #1011.

# Have

I have a Brother HL-L2325DW. I set it up as a driverless network printer through the GNOME settings using add printer and leaving defaults.

Running Fedora 40 (Workspace Edition) with GNOME on a Thinkpad T14s laptop. OpenPrinting CUPS 2.4.11.

# Problem 

Print jobs get canceled at printer repeatedly. Usually the first print after powering on the printer goes through, and anything else gets canceled.

# Tried

I've tried  to remove the printer and re-add, change settings, print from different places (lp, lpr, firefox, document viewer), factory reset printer, reset printer network settings.

# Logs

I enabled debug mode in `cups.conf`. Here is a session where the first document sent printed successfully, and the rest were canceled at printer. Search for `Unable to add document to print job.` which appears in red in my terminal. Sorry I cannot preserve the color coding. 

I hit the body limit so I ended up just putting the log in a repo because it was so long:

https://github.com/mcp292/printer-logs/tree/main

There's some odd stuff about "cropping" near the end.

These were set to print from Firefox.

**Comment by zdohnal** [comment]
Please file this as an issue - there might be others affected by it and they usually look into issues, not into discussion.

Github accepts .txt files as attachments in comments, so you don't need to add it as file in Github project.

Once you create the issue, please the log file with debug logs. Additionally, your current log shows an interesting event - it looks to me IPP backend reports there are mandatory job attributes, but we write them into PPD file as they are - which is incorrect and IMO it can cause the error after the first print.
So please put ```DirtyCleanInterval 0``` into cupsd.conf, restart the service, remove the printer, install it again and take the PPD file from ```/etc/cups/ppd```, add .txt and attach it to the issue - then print to the printer and once the printing finishes, take PPD from /etc/cups/ppd again, save it as another file and attach it as well.

**Reply by mcp292** [reply]
Thank you for the instructions and .txt tip. #1092 created.
